pkg/generator/appconfiguration/generators/job_generator.go

```diff
@@ -1,30 +1,97 @@
 package generators
 
 import (
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"kusionstack.io/kusion/pkg/models"
 	"kusionstack.io/kusion/pkg/models/appconfiguration/component"
 )
 
 type jobGenerator struct {
-	comp *component.Component
+	projectName string
+	compName    string
+	comp        *component.Component
 }
 
-func NewJobGenerator(comp *component.Component) (Generator, error) {
+func NewJobGenerator(projectName, compName string, comp *component.Component) (Generator, error) {
 	return &jobGenerator{
-		comp: comp,
+		projectName: projectName,
+		compName:    compName,
+		comp:        comp,
 	}, nil
 }
 
-func NewJobGeneratorFunc(comp *component.Component) NewGeneratorFunc {
+func NewJobGeneratorFunc(projectName, compName string, comp *component.Component) NewGeneratorFunc {
 	return func() (Generator, error) {
-		return NewJobGenerator(comp)
+		return NewJobGenerator(projectName, compName, comp)
 	}
 }
 
 func (g *jobGenerator) Generate(spec *models.Spec) error {
+	job := g.comp.Job
+	if job == nil {
+		return nil
+	}
+
 	if spec.Resources == nil {
 		spec.Resources = make(models.Resources, 0)
 	}
 
-	panic("unimplemented")
+	meta := metav1.ObjectMeta{
+		Namespace:   g.projectName,
+		Name:        uniqueComponentName(g.projectName, g.compName),
+		Labels:      g.comp.Labels,
+		Annotations: g.comp.Annotations,
+	}
+
+	containers, err := toOrderedContainers(job.Containers)
+	if err != nil {
+		return err
+	}
+	jobSpec := batchv1.JobSpec{
+		Template: corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: uniqueComponentLabels(g.projectName, g.compName),
+			},
+			Spec: corev1.PodSpec{
+				Containers: containers,
+			},
+		},
+	}
+
+	if job.Schedule == "" {
+		resource := &batchv1.Job{
+			ObjectMeta: meta,
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "Job",
+				APIVersion: "batch/v1",
```

Comment:
replace this hard code string with build in functions like `v1.SchemeGroupVersion.String()`